src/tls.c

```diff
@@ -8211,20 +8465,24 @@ static int server_generate_oqs_ciphertext(WOLFSSL* ssl,
             XFREE(keyShareEntry->ke, ssl->heap, DYNAMIC_TYPE_PUBLIC_KEY);
         }
 
-        keyShareEntry->pubKey = ciphertext;
-        keyShareEntry->pubKeyLen = (word32) kem->length_ciphertext;
-        ciphertext = NULL;
-
         keyShareEntry->ke = sharedSecret;
-        keyShareEntry->keLen = (word32) kem->length_shared_secret;
+        keyShareEntry->keLen = outlen + (word32)kem->length_shared_secret;
         sharedSecret = NULL;
+
+        XMEMCPY(ciphertext, ecc_kse->pubKey, ecc_kse->pubKeyLen);
+        keyShareEntry->pubKey = ciphertext;
+        keyShareEntry->pubKeyLen = (word32) ecc_kse->pubKeyLen +
```

Comment:
Clang on the Mac is complaining about loss of precision with this. `kem->length_ciphertext` is still 64-bit.